modules/redpanda/redpanda.go

```diff
@@ -75,38 +83,66 @@ func RunContainer(ctx context.Context, opts ...testcontainers.ContainerCustomize
 	// We have to do this kind of two-step process, because we need to know the mapped
 	// port, so that we can use this in Redpanda's advertised listeners configuration for
 	// the Kafka API.
-	entrypointFile, err := createEntrypointTmpFile()
-	if err != nil {
+	entrypointPath := filepath.Join(tmpDir, "entrypoint-tc.sh")
+	if err := os.WriteFile(entrypointPath, entrypoint, 0o700); err != nil {
 		return nil, fmt.Errorf("failed to create entrypoint file: %w", err)
 	}
 
 	// Bootstrap config file contains cluster configurations which will only be considered
 	// the very first time you start a cluster.
-	bootstrapConfigFile, err := createBootstrapConfigFile(settings)
+	bootstrapConfigPath := filepath.Join(tmpDir, ".bootstrap.yaml")
+	bootstrapConfig, err := renderBootstrapConfig(settings)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create bootstrap config file: %w", err)
 	}
+	if err := os.WriteFile(bootstrapConfigPath, bootstrapConfig, 0o600); err != nil {
+		return nil, fmt.Errorf("failed to create bootstrap config file: %w", err)
+	}
 
-	toBeMountedFiles := []testcontainers.ContainerFile{
-		{
-			HostFilePath:      entrypointFile.Name(),
+	req.Files = append(req.Files,
+		testcontainers.ContainerFile{
+			HostFilePath:      entrypointPath,
 			ContainerFilePath: "/entrypoint-tc.sh",
 			FileMode:          700,
 		},
-		{
-			HostFilePath:      bootstrapConfigFile.Name(),
+		testcontainers.ContainerFile{
+			HostFilePath:      bootstrapConfigPath,
 			ContainerFilePath: "/etc/redpanda/.bootstrap.yaml",
-			FileMode:          700,
+			FileMode:          600,
 		},
+	)
+
+	// 4. Create certificate and key for TLS connections.
+	if settings.EnableTLS {
+		certPath := filepath.Join(tmpDir, "cert.pem")
+		if err := os.WriteFile(certPath, settings.cert, 0o600); err != nil {
+			return nil, fmt.Errorf("failed to create certificate file: %w", err)
+		}
+		keyPath := filepath.Join(tmpDir, "key.pem")
+		if err := os.WriteFile(keyPath, settings.key, 0o600); err != nil {
+			return nil, fmt.Errorf("failed to create key file: %w", err)
+		}
+
+		req.Files = append(req.Files,
+			testcontainers.ContainerFile{
+				HostFilePath:      certPath,
+				ContainerFilePath: "/etc/redpanda/cert.pem",
```

Comment:
Wdyt about moving the paths to the cert files to constants at the top of the file? 🤔

Reply:
Have made all the file names constants. Better like that?